ygen/protogen.go

```diff
@@ -594,3 +668,57 @@ func genListKeyProto(list *yangDirectory, listName string, listPackage string, s
 
 	return km, nil
 }
+
+// enumInProtoUnionField parses an enum that is within a union and returns the generated
+// enumeration that should be included within a protobuf message for it.
+func enumInProtoUnionField(name string, types []*yang.YangType) (map[string]*protoMsgEnum, error) {
+	enums := map[string]*protoMsgEnum{}
+	for _, t := range types {
+		if t.Kind == yang.Yenum && t.Name == "enumeration" {
```

Comment:
isSimpleEnumType()

Reply:
Done - refactored since the helper was based on `yang.Entry`, should have taken your approach.